src/proxy-ui-api/frontend/src/views/Clients/ServiceClients/AddServiceClientServiceDialog.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+  <simpleDialog
+    :dialog="dialog"
+    :width="750"
+    title="serviceClients.addService"
+    @save="save"
+    @cancel="cancel"
+    :disableSave="filterSelections().length === 0"
+  >
+    <div slot="content" v-if="this.serviceCandidates.length > 0">
+      <v-text-field v-model="search"
+                    :label="$t('serviceClients.searchPlaceHolder')"
+                    single-line
+                    hide-details
+                    data-test="search-service-client"
+                    class="search-input">
+        <v-icon slot="append">mdi-magnify</v-icon>
+      </v-text-field>
+      <table class="xrd-table">
+        <thead>
+        <tr>
+          <th class="selection-checkbox"></th>
+          <th>{{$t('serviceClients.serviceCode')}}</th>
+          <th>{{$t('serviceClients.title')}}</th>
+        </tr>
+        </thead>
+        <tbody>
+        <tr v-for="accessRight in searchResults()"
+            v-bind:key="accessRight.id"
+            class="service-row"
+            data-test="access-right-toggle">
+          <td class="selection-checkbox">
+            <div>
+              <v-checkbox
+                v-model="selections"
+                :value="accessRight"
+                data-test="access-right-checkbox-input"
+              /></div>
+          </td>
+          <td>{{accessRight.service_code}}</td>
+          <td>{{accessRight.title}}</td>
+        </tr>
+        </tbody>
+      </table>
+    </div>
+    <div slot="content" v-else>
+      <h3>{{$t('serviceClients.noAvailableServices')}}</h3>
+    </div>
+  </simpleDialog>
+</template>
+<script lang="ts">
+  import Vue from 'vue';
+  import SimpleDialog from '@/components/ui/SimpleDialog.vue';
+  import {AccessRight} from '@/types';
+  export default Vue.extend({
+    components: {
+      SimpleDialog,
+    },
+    props: {
+      dialog: {
+        type: Boolean,
+        required: true,
+      },
+      serviceCandidates: {
+        type: Array,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        selections: [] as AccessRight[],
+        search: '' as string,
+      };
+    },
+    methods: {
+      save(): void {
+        const items = this.selections
+          .filter( (selection) => selection.service_code.includes(this.search))
+          .map( (selection): AccessRight => ({service_code: selection.service_code}));
+
+        this.$emit('save', items);
+        this.clear();
+      },
+      cancel(): void {
+        this.$emit('cancel');
+        this.clear();
+      },
+      clear(): void {
+        this.selections = [];
+      },
+      searchResults(): any {
```

Comment:
is there a reason to use any?

Reply:
also same one line down here

Reply:
serviceCandidates is a propvalue. I haven't found a working way to type them, is there such a way?

Reply:
Just marked the return type to ServiceCandidate[]

Reply:
Good question. I don't know a good way to do that either.

Reply:
You can use Prop<T> so this would be:
type: Array as Prop<ServiceCandidate[]>

Reply:
Hmm, I think I tried at some point. Don't remember how but it didn't work as well typings in data-block. Anyway I'll give it another shot...

Reply:
Ah I guess I tried something similar but it didn't work. But this does 👍